code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		slippery_rune(T)
+
+
+///just wet the turf under the rune every now and then, there's actually a small moment where the tile isn't wet
+/obj/effect/warped_rune/cyanspace/proc/slippery_rune(turf/open/T)
+	T.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+
+
+
+
+
+/*      Metal rune : makes an invisible wall. actually I lied, the rune is the wall.*/
+
+
+/obj/item/slimecross/warping/metal
+	colour = "metal"
+	runepath = /obj/effect/warped_rune/metalspace
+	effect_desc = "Creates a rune that prevents passage above it, takes longer to store and draw than other runes."
+	storing_time = 200  //storing time is longer for this particular rune
+	drawing_time = 100  //it's also longer to draw it
+
+//It's a wall what do you want from me
+/obj/effect/warped_rune/metalspace
+	desc = "Words are powerful things, they can stop someone dead in their tracks if used in the right way"
+	icon_state = "metal_space"
+	density = TRUE
+
+
+
+
+
+
+
+
+/*  Yellow rune space acts as an infinite generator, works without power and anywhere, recharges the APC of the room it's in and any batteries put on it
+
+*/
+
+
+/obj/item/slimecross/warping/yellow
+	colour = "yellow"
+	runepath = /obj/effect/warped_rune/yellowspace
+	effect_desc = "Creates a rune that Infinitely recharge batteries that are put on it, it will also passively recharge the APC of the room"
+
+
+/obj/effect/warped_rune/yellowspace
+	desc = "Where does all this energy come from? Who knows,the process does not matter, only the result."
+	icon_state = "elec_rune"
+
+
+/obj/effect/warped_rune/yellowspace/Initialize()
+	.=..()
+	START_PROCESSING(SSmachines, src)
+
+
+/obj/effect/warped_rune/yellowspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		apc_charge()
+		electrivore()
+
+/obj/effect/warped_rune/yellowspace/proc/apc_charge()
+
+	var/area/A = get_area(T)
+	for(var/obj/machinery/power/apc/P in A)
+		if(!P.cell)
+			return
+		if(P.cell.charge <! (P.cell.maxcharge - 1))
+			return
+		P.cell.charge += P.cell.maxcharge/100 //will basically recharge 1% of the APC cell every 10 seconds,we don't want to run engineers out of buisness
+		
+
+///charge whatever battery is put on the rune
+/obj/effect/warped_rune/yellowspace/proc/electrivore()
+	for(var/obj/item/stock_parts/cell/C in T)
+		if(C.charge < C.maxcharge) //check if the battery is not full
+			C.charge += C.maxcharge/10
+
+
+
+
+
+
+
+
+
+
+/* Dark purple crossbreed, creates a plasma crystal that can only be grinded for plasma dust, basically only useful to a xenobiologist  */
+
+
+/obj/item/slimecross/warping/darkpurple
+	colour = "dark purple"
+	runepath = /obj/effect/warped_rune/darkpurplespace
+	effect_desc = "Makes a rune that will periodically create pure plasma, if grinded it can produce a lot of plasma dust but that's it"
+
+
+/obj/effect/warped_rune/darkpurplespace
```

Comment:
alright, let's start Part III of my PR review

Instead of having this rune make plasma crystals, why not just have it add the plasma reagent directly to the contents of a random reagent container that's on its tile every time it processes? That way, these runes won't end up creating way more plasma crystals than any sane xenobiologist would reasonably need, and you don't need to micromanage a reagent grinder to make proper use of them.